src/test/osd/TestECCache.cc

```diff
@@ -0,0 +1,54 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2016 Western Digital Corporation.
+ *
+ * Author: Tomy Cheru <tomy.cheru@sandisk.com>
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ */
+
+#include <errno.h>
+#include <gtest/gtest.h>
+#include <osd/ECCache.hpp>
+#include <include/buffer.h>
+
+using namespace std;
+
+TEST(ecsc, basic_insert) {
+  ECCache ecsc(1000, 1000);
+
+  bufferlist bl;
+  char d[91] = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
+  bl.append(d, 90);
+
+  bufferlist tmp;
+
+  tmp.substr_of(bl, 30, 10);
+
+  bufferlist key;
+  key.substr_of(bl, 0, 10);
+  ecsc.put(key.c_str(), 30, tmp);
+
+  extents out;
+  uint64_t ret = ecsc.get(key.c_str(), 30, 10, &out);
+  EXPECT_NE(ret, (unsigned)0);
+  if(ret !=0) EXPECT_EQ(tmp, out.begin()->data());
+}
```

Comment:
I'd like to see a test here verifying the cache behavior under something like the following:

foo and bar are hobjects. x~y denotes offset x, length y (so from x to x+y).

tid 1 : start write foo 0~100
tid 2 : start write bar 50~100
tid 3 : start write foo 50~100
read foo 0~150 (assert correct contents, tid 1 from 0~50, tid 3 from 50~100)
tid 1 : finish write
read foo 0~150 (assert correct contents: nothing from 0~50, tid 3 from 50~100)
read bar 0~150 (assert correct contents: nothing from 0~50, tid 2 from 50~100)
etc